web3-dev/smart-contracts/getting-started.rst

```diff
@@ -5,61 +5,64 @@
 
-You have now npm project created with AssemblyScript installed among other dependencies.
-It will be used to generate bytecode from AssemblyScript code.
+  npx @massalabs/sc-project-initializer init my-sc && cd my-sc
 
-.. note::
-   Massa smart-contract module (`@massalabs/massa-as-sdk`) contains the API you need
-   to use to interact with the external world of the smart contract (the node, the ledger...).
+You now have an npm project, created in `my-sc` folder. It contains all tools that will be used to compile AssemblyScript to Wasm bytecode, as well as deploying and running your smart contracts.
 
 Congratulations! Now you have a fully set up project and you are ready to add some code.
 
 .. note::
    A few words on project folders:
 
-   * `assembly` is where the code goes;
+   * `assembly` is where the assemblyScript source code goes.
    * `build` will be created during compilation and will contain compiled smart contracts.
+   * `src` contains typescript code used to interact with the network.
+
 
 Create your first smart contract
 --------------------------------
 
 Since the beginning of mankind, humans explain how to use a program, a new language,
 a service by implementing a *Hello world!*. Your first smart contract will be no exception!
 
-Open the `main.ts` file in the `assembly` directory at the root of your project.
-Replace the code in the file by the following code:
+Open the `assembly/contracts/main.ts` file and replace its content by the following one:
 
 .. code-block:: typescript
 
-   import { generateEvent } from "@massalabs/massa-as-sdk";
+   // The entry file of your WebAssembly module.
+   import { Args } from '@massalabs/as-types';
+   import { generateEvent } from '@massalabs/massa-as-sdk';
 
-   export function main(_args: StaticArray<u8>): void {
-      generateEvent("Hello world!");
+   export function sayHello(args: StaticArray<u8>): void {
+      const name = new Args(args).nextString().unwrap();
+      const message = `Hello world! Greetings from ${name} 👋`;
+      generateEvent(message);
    }
 
+.. note::
+   Massa smart-contract module (`@massalabs/massa-as-sdk`) contains the API you need
+   to use to interact with the external world of the smart contract (the node, the ledger...).
+
 Don't forget to save the file. Before starting compilation, just a few words to describe what is used here:
 
-* line 1: `generateEvent` function is imported from Massa SDK (@massalabs/massa-as-sdk).
-  This function will generate an event with the string given as argument. Events can be later recovered using a Massa
-  client.
-* line 3: `main` function is exported. This means that the main function will be
-  callable from the outside of the WebAssembly module (more about that later).
-* line 4: `generateEvent` function is called with "Hello world!". Brian, we are thinking of you!
+* line 3: `sayHello` function is exported. This means that the `sayHello` function can be called
+  by another smart contract.
```

Comment:
```suggestion
  by the outside world (another smart contract of massa-web3).
```

Reply:
```suggestion
from outside of the smart contract. For instance by another smart contract or through the API (see :ref:`massa-web3 <web3-massa-web3>`).
```

Reply:
I think it is an important point that there are different ways to call a SC function. Probably mention it later when we show how to call it using the deployer.